source/server/server.cc

```diff
@@ -172,13 +172,23 @@ bool InstanceImpl::healthCheckFailed() { return server_stats_->live_.value() ==
 InstanceUtil::BootstrapVersion
 InstanceUtil::loadBootstrapConfig(envoy::config::bootstrap::v2::Bootstrap& bootstrap,
                                   Options& options) {
+  const std::string& config_path = options.configPath();
+  const std::string& config_yaml = options.configYaml();
+
+  // Exactly one of config_path and config_yaml should be specified.
+  if (config_path.empty() && config_yaml.empty()) {
+    const std::string message =
+        "At least one of --config-path and --config-yaml should be non-empty";
+    std::cerr << message << std::endl;
```

Comment:
As an aside, I wonder if main.cc should just print this directly from the exception message? I don't remember why we don't do that for other similar cases.

Reply:
Yeah that'd make sense. Added https://github.com/envoyproxy/envoy/issues/5453 for that.